duckencoder.py

```diff
@@ -22,160 +23,171 @@ def readResource(filename):
-                        val = splitted[1].strip()
-                        result_dict[key] = val
+                        key, _, val = l.partition("=")
+                        result_dict[key.strip()] = val.strip()
 
                 return result_dict
 
         @staticmethod
+        def altReadResource(filename):
+                with open(filename, "r") as f:
+                        lines = (line.strip().split("//")[0].partition("=") for line in f)
+                        return {line[0].strip(): line[-1].strip() for line in lines if line}
+
+        @staticmethod
         def parseScriptLine(line, keyProp, langProp):
                 result = ""
 
                 # split line into command and arguments
-                instr = line.split(" ", 1)
-                instr[0] = instr[0].strip()
-                if len(instr) > 1:
-                        instr[1] = instr[1].strip()
+                cmd, _, args = line.partition(" ")
+                cmd = cmd.strip()
+                args = args.strip()
 
                 # DELAY (don't check if second argument is present and int type)
-                if instr[0] == "DELAY":
-                        delay = int(instr[1])
+                if cmd == "DELAY":
+                        delay = int(args)
                         result = DuckEncoder.delay2USBBytes(delay)
 
                 # STRING
-                elif instr[0] == "STRING":
-                        if len(instr) == 1:
+                elif cmd == "STRING":
+                        if not args:
                                 return ""
                         # for every char
-                        for c in instr[1]:
+                        for c in args:
                                 keydata = DuckEncoder.ASCIIChar2USBBytes(c, keyProp, langProp)
                                 if len(keydata) > 0:
                                         result += keydata
 
                 # STRING_DELAY
-                elif instr[0] == "STRING_DELAY":
-                        if len(instr) < 2:
+                elif cmd == "STRING_DELAY":
+                        if not args:
                                 return ""
 
                         # split away delay argument from remaining string
-                        splitted = instr[1].split(" ", 1)
-                        splitted[0] = splitted[0].strip()
-                        splitted[1] = splitted[1].strip()
+                        delay, _, chars = args.partition(" ")
 
                         # build delaystr
-                        delay = int(splitted[0])
+                        delay = int(delay.strip())
                         delaystr = DuckEncoder.delay2USBBytes(delay)
 
                         # for every char
-                        chars = splitted[1]
-                        for c in chars:
+                        for c in chars.strip():
                                 keydata = DuckEncoder.ASCIIChar2USBBytes(c, keyProp, langProp)
```

Comment:
The `strip()` would disallow printing of trailing spaces I guess, but it is done the same way in the original implementation (https://github.com/hak5darren/USB-Rubber-Ducky/blob/master/Encoder/src/Encoder.java#L260). So yes, it should be done this way